pkg/envoy/lds/gateway.go

```diff
@@ -0,0 +1,84 @@
+package lds
+
+import (
+	"fmt"
+
+	xds_listener "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	"github.com/envoyproxy/go-control-plane/pkg/wellknown"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/envoy"
+	"github.com/openservicemesh/osm/pkg/service"
+)
+
+func (lb *listenerBuilder) buildGatewayListeners() []types.Resource {
+	if !lb.cfg.GetFeatureFlags().EnableMulticlusterMode {
+		return nil
+	}
```

Comment:
we can remove this check from here

Reply:
Let's keep this check here because I am going to use this flag and split the logic for ingress and multicluster appropriately.

Reply:
makes sense, thanks for clarifying